sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoPushTokenFetcher.kt

```
@@ -0,0 +1,27 @@
+/**
+ * [PushTokenFetcher] backed by Firebase Cloud Messaging.
+ */
+internal class KlaviyoPushTokenFetcher : PushTokenFetcher {
+    override fun fetchAndSetPushToken() {
+        try {
+            FirebaseMessaging.getInstance().token
+                .addOnSuccessListener { token -> Klaviyo.setPushToken(token) }
+                .addOnFailureListener { e ->
+                    Registry.log.warning("Failed to fetch push token for automatic registration", e)
+                }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)